### PR TITLE
made pennkeys lowercase and changed test case

### PR DESCRIPTION
**Pull request by nruia-penn**
Made it so when people send friend requests on penn courses, it changes the username to all lowercase so that when people add friends and accidentally capitalize the first letter it still is able to find the friend they are looking for.

### CODE REVIEW

File: backend/courses/views.py [hunk 1]
```diff
@@ -520,7 +520,8 @@ def get_all_friendships(self):
 
     def post(self, request):
         sender = request.user
-        recipient = get_object_or_404(User, username=request.data.get("pennkey"))
+
+        recipient = get_object_or_404(User, username=request.data.get("pennkey").lower())
```
Comment (Clue88) [inline backend/courses/views.py:524]: You should probably build in some logic here to handle the case where `request.data.get("pennkey")` returns `None`; this probably shouldn't happen based on input validation but we wouldn't want this to crash because of NoneType exception. (same with the other case that you modified)

You might also want to add some tests for unexpected inputs if they're not already there

**Review by Clue88 — CHANGES_REQUESTED** [review]
Looks good! Left a comment for some edge cases

**Comment by shiva-menta** [comment]
> Just made all pennkeys lowercase. Made a test for it, seems to work.

Would also be great to add 1-2 sentences on why this is needed! It's helpful to have more documentation on why we're making changes.

**Comment by nruia-penn** [comment]
Do you want me to add comments in the code or in some sort of documentation?

**Comment by shiva-menta** [comment]
> Do you want me to add comments in the code or in some sort of documentation?

Just within the PR itself would be sufficient. Something like [this](https://github.com/pennlabs/penn-courses/pull/667).

File: backend/courses/views.py [hunk 2]
```diff
@@ -520,7 +521,14 @@ def get_all_friendships(self):
 
     def post(self, request):
         sender = request.user
-        recipient = get_object_or_404(User, username=request.data.get("pennkey"))
+        res = {}
+
+        username = request.data.get("pennkey")
+        if not username:
+            res["message"] = "User not found"
+            return Response(res, status=status.HTTP_404_NOT_FOUND)
```
Comment (shiva-menta) [inline backend/courses/views.py:529]: Can make this more succinct and use `raise Http404("User not found.")`.

File: backend/courses/views.py [hunk 3]
```diff
@@ -554,9 +562,17 @@ def post(self, request):
...
+
         sender = request.user
-        recipient = get_object_or_404(User, username=request.data.get("pennkey"))
+
+        res = {}
+
+        username = request.data.get("pennkey")
+        if not username:
+            res["message"] = "User not found."
+            return Response(res, status=status.HTTP_404_NOT_FOUND)
```
Comment (shiva-menta) [inline backend/courses/views.py:573]: Same thing here.

**Review by shiva-menta — APPROVED** [review]
lgtm – thanks Nikhil!

**Review by Clue88 — APPROVED** [review]
looks great!
